src/rust/src/expr/string.rs

```diff
@@ -0,0 +1,382 @@
+    //             exact,
+    //             cache,
+    //         })
+    //         .into())
+    // }
+
+    // #[allow(clippy::too_many_arguments)]
+    // fn str_to_datetime(
+    //     &self,
+    //     format: &PlRExpr,
+    //     time_unit: &PlRExpr,
+    //     time_zone: &PlRExpr,
+    //     strict: &PlRExpr,
+    //     exact: &PlRExpr,
+    //     cache: &PlRExpr,
+    //     ambiguous: &PlRExpr,
+    // ) -> Result<Self> {
+    //     let format = robj_to!(Option, String, format)?.map(|x| x.into());
+    //     let time_unit = robj_to!(Option, timeunit, time_unit)?.map(|x| x.into());
+    //     let time_zone = robj_to!(Option, String, time_zone)?.map(|x| x.into());
+
+    //     Ok(self
+    //         .inner
+    //         .clone()
+    //         .str()
+    //         .to_datetime(
+    //             time_unit,
+    //             time_zone,
+    //             StrptimeOptions {
+    //                 format: format,
+    //                 strict: robj_to!(bool, strict)?,
+    //                 exact: robj_to!(bool, exact)?,
+    //                 cache: robj_to!(bool, cache)?,
+    //             },
+    //             robj_to!(PLExpr, ambiguous)?,
+    //         )
+    //         .into())
+    // }
+
+    // fn str_to_time(&self, format: &PlRExpr, strict: &PlRExpr, cache: &PlRExpr) -> Result<Self> {
+    //     let format = robj_to!(Option, String, format)?.map(|x| x.into());
+
+    //     Ok(self
+    //         .inner
+    //         .clone()
+    //         .str()
+    //         .to_time(StrptimeOptions {
+    //             format,
+    //             strict: robj_to!(bool, strict)?,
+    //             cache: robj_to!(bool, cache)?,
+    //             exact: true,
+    //         })
+    //         .into())
+    // }
+
+    fn str_split(&self, by: &PlRExpr, inclusive: bool) -> Result<Self> {
+        if inclusive {
+            Ok(self
+                .inner
+                .clone()
+                .str()
+                .split_inclusive(by.inner.clone())
+                .into())
+        } else {
+            Ok(self.inner.clone().str().split(by.inner.clone()).into())
+        }
+    }
+
+    fn str_split_exact(&self, by: &PlRExpr, n: usize, inclusive: bool) -> Result<Self> {
+        Ok(if inclusive {
+            self.inner
+                .clone()
+                .str()
+                .split_exact_inclusive(by.inner.clone(), n)
+        } else {
+            self.inner.clone().str().split_exact(by.inner.clone(), n)
+        }
+        .into())
+    }
+
+    fn str_splitn(&self, by: &PlRExpr, n: usize) -> Result<Self> {
+        Ok(self.inner.clone().str().splitn(by.inner.clone(), n).into())
+    }
+
+    fn str_replace(&self, pat: &PlRExpr, value: &PlRExpr, literal: bool, n: i32) -> Result<Self> {
+        Ok(self
+            .inner
+            .clone()
+            .str()
+            .replace_n(pat.inner.clone(), value.inner.clone(), literal, n as i64)
+            .into())
+    }
```

Comment:
Instead of conversion by `as`, convert from `NumericScalar` via `<Wrap<i64>>::try_from()` as follows.
https://github.com/eitsupi/neo-r-polars/blob/d5994dc6a55d0cdea27a77eeb54fa04be701c1fd/src/rust/src/dataframe/general.rs#L55-L65

Do not use `as.integer()` on the R side because it may cause unexpected conversions.